zone/spells.cpp

```diff
@@ -4544,39 +4544,62 @@ bool Mob::SpellOnTarget(
 	return true;
 }
 
-void Corpse::CastRezz(uint16 spellid, Mob* Caster)
+void Corpse::CastRezz(uint16 spell_id, Mob* Caster)
 {
-	LogSpells("Corpse::CastRezz spellid [{}], Rezzed() is [{}], rezzexp is [{}]", spellid,IsRezzed(),rez_experience);
-
-	if(IsRezzed()){
-		if(Caster && Caster->IsClient())
-			Caster->Message(Chat::Red,"This character has already been resurrected.");
-
+	LogSpells("Corpse::CastRezz spell_id [{}], Rezzed() is [{}], rezzexp is [{}], rez_timer enabled:: [{}]", spell_id,IsRezzed(),rez_experience, corpse_rez_timer.Enabled());
+
+	// refresh rezzed state from database
+	uint32 db_exp, db_gm_exp;
+	bool db_rezzable, db_is_rezzed;
+	if (!database.LoadCharacterCorpseRezData(corpse_db_id, &db_exp, &db_gm_exp, &db_rezzable, &db_is_rezzed)) {
```

Comment:
Instead of making a function for this and passing all of this by reference, you can do it inline (remove the function)

```cpp
const auto& e = CharacterCorpsesRepository::FindOne(*this, corpse_id);
if (!e.id) {
	return false;
}

if (!e.id) {
	Caster->MessageString(Chat::White, REZZ_ALREADY_PENDING);
	return;
}

db_exp       = e.exp;
db_gm_exp    = e.gm_exp;
db_rezzable  = e.rezzable;
db_is_rezzed = e.is_rezzed;
```

Also the `db_` prefixes can be removed